workflow-manager/limiter/limiter_test.go

```diff
@@ -0,0 +1,42 @@
+package limiter
+
+import (
+	"testing"
+	"time"
+)
+
+func TestTicketDoubleDone(t *testing.T) {
+	l := New(5)
+	l.Execute(func(ticket *Ticket) {
+		l.Done(ticket)
+		l.Done(ticket)
+	})
+	t.Log("l.Execute did not block")
```

Comment:
How does a failure in this test manifest? The Go deadlock detector will eventually fire and abort the test? Does that let the rest of the test suite go through? I think it'd be better to do something like:
```
l := New(5)
doneChan := make(chan struct{})
go l.Execute(func(ticket *Ticket) {
    l.Done(ticket)
    l.Done(ticket)
    doneChan <- struct{}{}
})
for {
    select {
        case <-doneChan:
            // success
            return
        case <-time.After(time.Second):
            t.Fatal("timeout waiting for job to complete")
    }
}
```